### PR TITLE
docs: Added detailed explanation on fleet's Metadata logic

### PR DESCRIPTION
**Pull request by kayrus**
/cc @joshix @robszumski @jonboulle 
resolves https://github.com/coreos/fleet/issues/1411

### CODE REVIEW

File: Documentation/unit-files-and-scheduling.md [hunk 1]
```diff
@@ -104,29 +98,51 @@ One must use the entire ID when setting `MachineID` - the shortened ID returned
...
 The `MachineMetadata` option of a unit file allows you to set conditional metadata required for a machine to be elegible.
 
-```
+```ini
 [X-Fleet]
 MachineMetadata="region=us-east-1" "diskType=SSD"
 ```
 
-This requires an eligible machine to have at least the `region` and `diskType` keys set accordingly. A single key may also be defined multiple times, in which case only one of the conditions needs to be met:
+This requires an eligible machine to have at least the `region` and `diskType` keys set accordingly. I.e. this logic could be represented as follows:
```
Comment (jonboulle) [inline Documentation/unit-files-and-scheduling.md]: s#I.e.#This logic#

File: Documentation/unit-files-and-scheduling.md [hunk 2]
```diff
@@ -104,29 +98,51 @@ One must use the entire ID when setting `MachineID` - the shortened ID returned
...
+
+If we combine two previous examples in one:
+
+```ini
+[X-Fleet]
+MachineMetadata="region=us-east-1" "diskType=SSD"
+MachineMetadata=region=us-west-1
+```
+
+This logic would be as follows:
```
Comment (jonboulle) [inline Documentation/unit-files-and-scheduling.md]: s/This/the/

File: Documentation/unit-files-and-scheduling.md [hunk 3]
```diff
@@ -104,29 +98,51 @@ One must use the entire ID when setting `MachineID` - the shortened ID returned
...
+
+```sql
+diskType=SSD AND (region=us-east-1 OR region=us-west-1)
```
Comment (jonboulle) [inline Documentation/unit-files-and-scheduling.md]: you've verified this, right? :-)

**Comment by jonboulle** [comment]
A few small comments, lgtm after that

**Comment by kayrus** [comment]
@jonboulle update commit and added more details.

File: Documentation/unit-files-and-scheduling.md [hunk 4]
```diff
@@ -104,29 +98,74 @@ One must use the entire ID when setting `MachineID` - the shortened ID returned
...
+MachineMetadata=region=us-west-1
+```
+
+the logic would be as follows:
+
+```sql
+diskType=SSD AND (region=us-east-1 OR region=us-west-1)
+```
+
+Previous example schedules maximum one unit accross your cluster depending on first met requirement. If you add `Global=true`:
```
Comment (jonboulle) [inline Documentation/unit-files-and-scheduling.md]: The previous example schedules at most one unit across your cluster, depending on the first satisfied requirement.

**Comment by jonboulle** [comment]
One more comment then looks good to merge.

**Comment by kayrus** [comment]
@jonboulle fixed

**Pull request merged**